src/Task/Runner/TaskRunner.php

```diff
@@ -127,7 +127,7 @@ private function handle(TaskExecutionInterface $execution)
         );
 
         try {
-            return $handler->handle($execution->getWorkload());
+            return $handler->handle($execution->getWorkload(), $execution);
```

Comment:
This would lead into an "optional" argument? I would prefer a solution which the execution exists also in the interface. it is a BC-break but ok when you change this PR to develop.